### ISSUE TITLE
P1 Corner Lightstrips

### ISSUE THREAD

**Issue by Cosmiscient** [BEEmod/BEE2-items#3737]
**Description of the feature:** Lightstrips in P1 will change after compilation if lightstrips are:
- in p1 style
- placed on the ceiling against a wall / block
- placed in a full row across the ceiling until it hits a wall / block

**This should be implemented because:** It would be near impossible to misuse this very specific feature, and would be a really helpful, impressive way to light up a wall.

**Additional information:**
![image](https://user-images.githubusercontent.com/42079031/113764179-9fa67f00-96e8-11eb-9cd5-4bd53d4dfb77.png)
![image](https://user-images.githubusercontent.com/42079031/113764239-b1882200-96e8-11eb-853b-d4fa4c949802.png)

**Comment by TeamSpen210** [comment]
This would be good, but it's quite difficult to generate these strips.

**Comment by Lieutenant-L-T-Smash** [comment]
I'm not aware of all the background technical details, but it seems to me this could be implemented in a  similar fashion to the existing P1 hole lights. The "hole" is just very thin, and long enough to span the boxel. It would carve into the adjacent boxel the way the square lights do.

There is also no real need to restrict placement to the edge of the room, or overload one of the existing light strip types to change based on context. This thin kind of light strip can have its own dedicated type (Sphere-type is free) and while it wouldn't necessarily look good if placed away from a wall, it's on the map designer to use the freedom responsibly.

**Comment by Konclan** [comment]
Because these are exclusively specific to p1 style and you can't really stack light holes adding a new item just for this wouldn't be worth it unless we had some ideas for the other styles.

**Comment by Luke18033** [comment]
Yeah, they're kind of a difficult thing to implement, but they're also important to get the accurate P1 look.
- Placing light strips along the floor next to a wall makes corners impossible, since the light strips can't overlap.
- TeamSpen had the idea to have you place them along the *walls* next to the *floor* instead, that would allow corners to work since they no longer need to overlap. But then that prevents you from having most other items on those walls. Maybe you'd just remove the light strips where you want to put other items, then we use a bit of compiler logic to fill in those gaps.
- We could make the edge lights a separate item, which would allow them to be put on the floor again since we can disable collisions, but then what should they look like in other styles? I guess we could add style variants:
	- Clean and Overgrown could still use light hole designs, but maybe change the color a bit.
	- Old Aperture could use thin fluorescent light strips mounted onto the floor (though still nonsolid).

Separate item seems like the best option, although then we have to figure out what to do for the style variants.

**Comment by The-Queen55142** [comment]
Edge lights could do something similar to the Vactubes or Catwalks, where you connect one end to another to signal the compiler to add them in. (See image)
![example](https://user-images.githubusercontent.com/37309250/139942080-ba9ed36d-08b9-4a3e-9932-9ee25a151130.png)

Clean could use a similar design to the P1 design but with altered lighting, as stated above*
Overgrown could have some plants, dirt, or whatever sprinkled ontop of the lights, causing the lights to be a slight bit less effective (turn brightness down), and occasional randomly generated broken lights, potentially letting you see a piston for one of the panels (due to all tests in Portal 2 being formed out of panels)

Old Aperture could resemble those bar lights you can buy online, the long strip lights often used on the underside of bar counters? It would fit snug in the corner between the floor and walls., The light coloration(s) would be consistent with Old Aperture's Light Strips.*



Art Therapy could have a more bare bones version of the Old Aperture version, perhaps having a random chance to have a wire visible or a light missing, since the style is sorta mashed together. It could match the color given off by other lights in Art Therapy (according to Valve's developer site.)*

***Alternatively, a cube type for warm, cool, and neutral.**

**Comment by Konclan** [comment]
Actually it might be better to add this to automatic interior chamber decor. These lights are really common in most Portal 1 chambers and they would rarely obstruct portals. If they do people could just use the item that prevents decorations.

**Comment by TeamSpen210** [comment]
Yep, it also has positioning requirements due to it embedding, so we'd want the compiler to just skip if not possible.

**Comment by Luke18033** [comment]
I think lighting items like this should always be manually placeable... Maybe you place two items at the corners and connect them like a cutout tile, then it'll automatically detect edges within that area and add the lights there.

**Comment by Luke18033** [comment]
Sort of like TeamSpen suggested for handrails in #1274, just detecting walls instead of ledges.
